modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/source/poll/PollingSourceWrapper.java

```
@@ -235,6 +243,20 @@ private void poll(SourceCallback<T, A> sourceCallback) {
     });
   }
 
+  private boolean shouldSkipPoll() {
+    if (sequentialPolls) {
+      try {
+        return inflightIdsObjectStore.allKeys().size() != 0;
+      } catch (ObjectStoreException e) {
+        LOGGER
+            .error(format("Skipping poll on flow '%s', because there was an error trying to verify if the items from last poll have finished processing. %s",
```

don't hide exception